dev/TEAL_opcodes.md

```diff
@@ -1046,339 +959,352 @@ params: Before v4, Txn.ForeignAssets offset. Since v4, Txn.ForeignAssets offset
+`itxn_begin` initializes Sender to the application address; Fee to the minimum allowable, taking into account MinTxnFee and credit from overpaying in earlier transactions; FirstValid/LastValid to the values in the invoking transaction, and all other fields to zero or empty values.
 
 ## itxn_field f
 
 - Opcode: 0xb2 {uint8 transaction field index}
-- Pops: *... stack*, any
-- Pushes: _None_
-- set field F of the current inner transaction to X
-- LogicSigVersion >= 5
+- Stack: ..., A &rarr; ...
+- set field F of the current inner transaction to A
+- Availability: v5
 - Mode: Application
 
-`itxn_field` fails if X is of the wrong type for F, including a byte array of the wrong size for use as an address when F is an address field. `itxn_field` also fails if X is an account or asset that does not appear in `txn.Accounts` or `txn.ForeignAssets` of the top-level transaction. (Setting addresses in asset creation are exempted from this requirement.)
+`itxn_field` fails if A is of the wrong type for F, including a byte array of the wrong size for use as an address when F is an address field. `itxn_field` also fails if A is an account, asset, or app that is not _available_. (Addresses set into asset params of acfg transactions need not be _available_.)
 
 ## itxn_submit
 
 - Opcode: 0xb3
-- Pops: _None_
-- Pushes: _None_
-- execute the current inner transaction. Fail if 16 inner transactions have already been executed, or if the transaction itself fails.
-- LogicSigVersion >= 5
+- Stack: ... &rarr; ...
+- execute the current inner transaction group. Fail if executing this group would exceed the inner transaction limit, or if any transaction in the group fails.
+- Availability: v5
 - Mode: Application
 
 `itxn_submit` resets the current transaction so that it can not be resubmitted. A new `itxn_begin` is required to prepare another inner transaction.
 
 ## itxn f
 
 - Opcode: 0xb4 {uint8 transaction field index}
-- Pops: _None_
-- Pushes: any
-- push field F of the last inner transaction to stack
-- LogicSigVersion >= 5
+- Stack: ... &rarr; ..., any
+- field F of the last inner transaction
+- Availability: v5
 - Mode: Application
 
 ## itxna f i
 
 - Opcode: 0xb5 {uint8 transaction field index} {uint8 transaction field array index}
-- Pops: _None_
-- Pushes: any
-- push Ith value of the array field F of the last inner transaction to stack
-- LogicSigVersion >= 5
+- Stack: ... &rarr; ..., any
+- Ith value of the array field F of the last inner transaction
+- Availability: v5
+- Mode: Application
+
+## itxn_next
+
+- Opcode: 0xb6
+- Stack: ... &rarr; ...
+- begin preparation of a new inner transaction in the same transaction group
+- Availability: v6
+- Mode: Application
+
+`itxn_next` initializes the transaction exactly as `itxn_begin` does
```

Comment:
Just confirming that `itxn_next` never does any check and can never fail.
Maybe to be added?

Reply:
It can fail if you hit a limit on the total number of allowable inner transactions, or if the group your building exceeds 16.